### PR TITLE
Add `screenRecordingPermission` option

### PR DESCRIPTION
**Pull request by wrgoto**
### Why

One of the significant drawbacks is the library asking for Screen Recording permissions for macOS versions 10.15 and newer. Even though the library doesn't actually record the screen, it is required to get the title of the active window. For those who do not need the window title, there should be an option to forgo the Screen Recording permission check.

### Changes

- Add a `disableScreenRecordingPermission` option.
- Skip the screen recording permission check if true.
- Return an empty string as the title if true.

### Related

- https://github.com/sindresorhus/active-win/issues/88
- https://github.com/sindresorhus/active-win/pull/67

### CODE REVIEW

**Comment by sindresorhus** [comment]
See the feedback in https://github.com/sindresorhus/active-win/pull/67/files. There should be an option for each check.

File: index.d.ts [hunk 1]
```diff
@@ -1,4 +1,13 @@
 declare namespace activeWin {
+	interface Options {
+		/**
+		Disable the screen recording permission check (macOS)
+
+		Setting this to true will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property will always be set to an empty string.
+		 */
+		disableScreenRecordingPermission: boolean;
```
Comment (sindresorhus) [inline index.d.ts:8]:
```suggestion
		screenRecordingPermission: boolean;
```

File: index.d.ts [hunk 2]
```diff
@@ -1,4 +1,13 @@
 declare namespace activeWin {
+	interface Options {
+		/**
+		Disable the screen recording permission check (macOS)
+
+		Setting this to true will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property will always be set to an empty string.
```
Comment (sindresorhus) [inline index.d.ts:6]: Should mention the default value with `@default`.

https://github.com/sindresorhus/typescript-definition-style-guide

File: readme.md [hunk 3]
```diff
@@ -45,13 +45,17 @@ const activeWin = require('active-win');
...
 
-### activeWin.sync()
+### activeWin.sync(options)
 
-Returns an `Object` with the result, or `undefined` if there is no active window.
+Accepts an optional dictionary `Options` as only argument and returns an `Object` with the result, or `undefined` if there is no active window.
+
+## Options
+
+- `disableScreenRecordingPermission` *(boolean)* - Disable the screen recording permission check (macOS). Setting this to true will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property will always be set to an empty string.
```
Comment (sindresorhus) [inline readme.md:58]: Should use formatting like https://github.com/sindresorhus/boxen#options

File: readme.md [hunk 4]
```diff
@@ -45,13 +45,17 @@ const activeWin = require('active-win');
 
 ## API
 
-### activeWin()
+### activeWin(options)
```
Comment (sindresorhus) [inline readme.md:48]:
```suggestion
### activeWin(options?)
```

**Comment by achuinard** [comment]
@wrgoto Thanks for your contributions!

**Comment by wrgoto** [comment]
@sindresorhus I agree there should be an option for each check, but I haven't verified the extent of removing the accessibility check. I'm not entirely sure if it is limited to the `url` property. I can investigate in a separate issue and PR.

**Comment by achuinard** [comment]
@sindresorhus are you looking to get this merged soon?

File: index.test-d.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import {Result, LinuxResult, MacOSResult, WindowsResult} from '.';
 
 expectType<Promise<Result | undefined>>(activeWin());
 
-const result = activeWin.sync();
+const result = activeWin.sync({ screenRecordingPermission: false });
```
Comment (sindresorhus) [inline index.test-d.ts:7]:
```suggestion
const result = activeWin.sync({screenRecordingPermission: false});
```

File: lib/macos.js [hunk 6]
```diff
@@ -19,9 +19,25 @@ const parseMac = stdout => {
...
-module.exports = async () => {
-	const {stdout} = await execFile(bin);
+const getArguments = options => {
+	if (!options) {
+		return [];
+	}
+
+	const args = [];
+	if (options.screenRecordingPermission === false) {
+		args.push('--disable-screen-recording-permission');
```
Comment (sindresorhus) [inline lib/macos.js:29]:
```suggestion
		args.push('--no-screen-recording-permission');
```

`--no` is the convention for inverted flags.

File: readme.md [hunk 7]
```diff
@@ -45,16 +45,36 @@ const activeWin = require('active-win');
...
+#### options
+
+Type: `object`
+
+##### screenRecordingPermission
+
+Type: `boolean`\
+Default: `true`
+
+Enable the screen recording permission check (macOS). Setting this to `false` will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property in the result will always be set to an empty string.
```
Comment (sindresorhus) [inline readme.md:59]:
```suggestion
Enable the screen recording permission check. Setting this to `false` will prevent the screen recording permission prompt on macOS versions 10.15 and newer. The `title` property in the result will always be set to an empty string.
```

File: readme.md [hunk 8]
```diff
@@ -45,16 +45,36 @@ const activeWin = require('active-win');
...
 
 ## Result
 
+Returns a `Promise<Object>` with the result, or `Promise<undefined>` if there is no active window or if the information is not available.
```
Comment (sindresorhus) [inline readme.md:76]:
```suggestion
Returns a `Promise<object>` with the result, or `Promise<undefined>` if there is no active window or if the information is not available.
```

File: readme.md [hunk 9]
```diff
@@ -45,16 +45,36 @@ const activeWin = require('active-win');
...
 
-### activeWin()
+### activeWin(options?)
 
-Returns a `Promise<Object>` with the result, or `Promise<undefined>` if there is no active window or if the information is not available.
+#### options
+
+Type: `object`
+
+##### screenRecordingPermission
```
Comment (sindresorhus) [inline readme.md:54]:
```suggestion
##### screenRecordingPermission **(macOS only)**
```

File: readme.md [hunk 10]
```diff
@@ -45,16 +45,36 @@ const activeWin = require('active-win');
...
+
+Type: `boolean`\
+Default: `true`
+
+Enable the screen recording permission check (macOS). Setting this to `false` will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property in the result will always be set to an empty string.
+
+### activeWin.sync(options?)
 
-### activeWin.sync()
+#### options
```
Comment (sindresorhus) [inline readme.md:63]: You don't need to document the `options` twice.

File: index.d.ts [hunk 11]
```diff
@@ -1,4 +1,15 @@
 declare namespace activeWin {
+	interface Options {
+		/**
+		Enable the screen recording permission check (macOS).
```
Comment (sindresorhus) [inline index.d.ts:4]:
```suggestion
		Enable the screen recording permission check. _(macOS)_
```

File: index.d.ts [hunk 12]
```diff
@@ -1,4 +1,15 @@
...
+
+		Setting this to `false` will prevent the Screen Recording permission prompt on macOS versions 10.15 and newer. The `title` property in the result will always be set to an empty string.
```
Comment (sindresorhus) [inline index.d.ts:6]:
```suggestion
		Setting this to `false` will prevent the screen recording permission prompt on macOS versions 10.15 and newer. The `title` property in the result will always be set to an empty string.
```

File: index.d.ts [hunk 13]
```diff
@@ -1,4 +1,15 @@
...
+
+		@default true
+		*/
+		screenRecordingPermission: boolean;
```
Comment (sindresorhus) [inline index.d.ts:10]:
```suggestion
		readonly screenRecordingPermission: boolean;
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
The PR title needs to be updated.

**Pull request merged**